File: frontend/src/services/reviewService.js
import axiosClient from "../api/axiosClient";

const BASE_URL = "/api/review";

const reviewService = {

    saveReview: async (review) => {

        const response = await axiosClient.post(
            BASE_URL,
            review
        );

        return response.data.data;

    },

    getAllReviews: async () => {

        const response = await axiosClient.get(
            `${BASE_URL}/all`
        );

        return response.data.data;

    },

    getUserReviews: async (userId) => {

        const response = await axiosClient.get(
            `${BASE_URL}/user/${userId}`
        );

        return response.data.data;

    },

    getTechnicianReviews: async (technicianId) => {

        const response = await axiosClient.get(
            `${BASE_URL}/technician/${technicianId}`
        );

        return response.data.data;

    }

};

export default reviewService;

File: src/main/java/com/csms/services/ReviewService.java
package com.csms.services;

import java.util.List;
import java.util.Optional;

import com.csms.dto.ReviewDto;
import com.csms.entity.Review;

public interface ReviewService {

    // User submits a review
    Optional<Review> save(ReviewDto dto);

    // Admin - View all reviews
    List<ReviewDto> getAllReviews();

    // User - View own reviews
    List<ReviewDto> getReviewsByUser(Integer userId);

    // Technician - View reviews received
    List<ReviewDto> getReviewsByTechnician(Integer technicianId);

}